doc/source/methods/modeldistillation.ipynb

```diff
@@ -0,0 +1,180 @@
+    "distilled_model = tf.keras.Sequential(\n",
+    "    [\n",
+    "        tf.keras.InputLayer(input_shape=(input_dim,)),\n",
+    "        tf.keras.layers.Dense(output_dim, activation=tf.nn.softmax)\n",
+    "    ]\n",
+    ")\n",
+    "```\n",
+    "\n",
+    "* `model`: the classifier as a `tf.keras.Model`. Example:\n",
+    "\n",
+    "```python\n",
+    "inputs = tf.keras.Input(shape=(input_dim,))\n",
+    "hidden = tf.keras.layers.Dense(hidden_dim)(inputs)\n",
+    "outputs = tf.keras.layers.Dense(output_dim, activation=tf.nn.softmax)(hidden)\n",
+    "model = tf.keras.Model(inputs=inputs, outputs=outputs)\n",
+    "```\n",
+    "\n",
+    "* `loss_type`: type of loss used for distillation. Supported losses: 'kld', 'xent'.\n",
+    "\n",
+    "* `temperature`: Temperature used for model prediction scaling. Temperature <1 sharpens the prediction probability distribution which can be beneficial for prediction distributions with high entropy.\n",
+    "\n",
+    "* `data_type`: can specify data type added to metadata. E.g. *'tabular'* or *'image'*.\n",
+    "\n",
+    "Initialized detector example:\n",
+    "\n",
+    "```python\n",
+    "from alibi_detect.ad import ModelDistillation\n",
+    "\n",
+    "ad = AdversarialAE(\n",
+    "    distilled_model=distilled_model,\n",
+    "    model=model,\n",
+    "    temperature=0.5\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fit\n",
+    "\n",
+    "We then need to train the detector. The following parameters can be specified:\n",
+    "\n",
+    "* `X`: training batch as a numpy array.\n",
+    "\n",
+    "* `loss_fn`: loss function used for training. Defaults to the custom adversarial loss.\n",
+    "\n",
+    "* `optimizer`: optimizer used for training. Defaults to [Adam](https://arxiv.org/abs/1412.6980) with learning rate 1e-3.\n",
+    "\n",
+    "* `epochs`: number of training epochs.\n",
+    "\n",
+    "* `batch_size`: batch size used during training.\n",
+    "\n",
+    "* `verbose`: boolean whether to print training progress.\n",
+    "\n",
+    "* `log_metric`: additional metrics whose progress will be displayed if verbose equals True.\n",
+    "\n",
+    "* `preprocess_fn`: optional data preprocessing function applied per batch during training.\n",
+    "\n",
+    "\n",
+    "```python\n",
+    "ad.fit(X_train, epochs=50)\n",
+    "```\n",
+    "\n",
+    "The threshold for the adversarial / harmfulness score can be set via ```infer_threshold```. We need to pass a batch of instances $X$ and specify what percentage of those we consider to be normal via `threshold_perc`. Even if we only have normal instances in the batch, it might be best to set the threshold value a bit lower (e.g. $95$%) since the the model could have misclassified training instances leading to a higher score if the reconstruction picked up features from the correct class or some instances might look adversarial in the first place.\n",
```

Comment:
Remove "leading to a higher score if the reconstruction picked up features from the correct class" b/c it relates to the adversarial AE.